documentation/tutorials/get-started-with-ash-and-phoenix.md

```diff
@@ -0,0 +1,501 @@
+    # Add an autogenerated UUID primary key called `:id`.
+    uuid_primary_key :id
+    # Add a string type attribute called `:title`
+    attribute :title, :string do
+      # We don't want the title to ever be `nil`
+      allow_nil? false
+    end
+
+    # Add a string type attribute called `:content`
+    # If allow_nil? is not specified, then content can be nil
+    attribute :content, :string
+  end
+end
+```
+
+### Migrate the Database
+
+We have specified the resource in Ash. But we have yet to create it in our data layer (in our case Postgres).
+
+We created our database earlier, but now we need to populate it. We do this by generating and performing a migration.
+
+We can use a generator to produce a migration for us. Ash can deduce what needs to go into the migration and do the hard work for us, to do this use the command below:
+
+```bash
+$ mix ash_postgres.generate_migrations --name initial_migration
+
+# ... don't worry about other files it creates
+
+Generating Migrations:
+* creating priv/repo/migrations/20230208045101_initial_migration.exs
+```
+
+Here is the migration file commented in detail:
+
+```elixir
+defmodule MyAshPhoenixApp.Repo.Migrations.InitialMigration do
+  use Ecto.Migration
+
+  # This function runs when migrating forward
+  def up do
+    # Creates the `:posts` table
+    create table(:posts, primary_key: false) do
+      # Adds primary key attribute `:id` of type `:uuid`
+      # null values are not allowed
+      add :id, :uuid, null: false, default: fragment("uuid_generate_v4()"), primary_key: true
+
+      # Adds attribute `:title` of type `:text`, null values are not allowed
+      add :title, :text, null: false
+      # Adds attribute `:content` of type `:text`, null values are allowed
+      add :content, :text
+    end
+  end
+
+  # This is the function that runs if you want to rollback the migration.
+  def down do
+    # Deletes the `:posts` table
+    drop table(:posts)
+  end
+end
+```
+
+We can run the `up/0` function which will perform the desired operations on the Postgres database. We do this with the migrate command:
+
+```bash
+$ mix ash_postgres.migrate
+```
+
+## Interacting with your Resources
+
+**All interaction with your resource attributes always occur through an action**. In our resource we are using the default actions for `:create, :read, :update, :destroy` along with a custom action `:by_id`.
+
+`:create` and `:update` and `:destroy` actions require a changeset. Ash changesets are conceptually similar to [Ecto changesets](https://hexdocs.pm/ecto/Ecto.Changeset.html). They're data structures which represent an intended change to an Ash resource and provide validation.
+
+The `:read` action takes a query instead of a changeset
+
+Below is the most verbose way of calling your resource. All other ways of interaction is some kind of shorthand of these. This means at some point a changeset is being created and passed to the API, even if its encapsulated within another function.
```

Comment:
`it's encapsulated`

Reply:
[675a3a4](https://github.com/ash-project/ash/pull/504/commits/675a3a41730b743fd0bd8b5db419557a0068ff22)